### PR TITLE
feat(dialog): add component - INNO-679

### PR DESCRIPTION
**Pull request by kalinchernev**
# PR description

Named dialog to be closer to the web specifications.
To test, please add `<a href="#dialog" class="ecl-link" data-ecl-dialog="ecl-dialog">Open dialog</a>` or similar trigger in the template to be able to run the code.

I feel this part can be better, I welcome suggestions whether should I rework a bit this from the other component. 
```
<button type="button" class="ecl-message__dismiss" aria-label="{{ dismiss.label }}">&times;</button>
``` 

I haven't included tests, as there is nothing visual without adding an actual trigger.

As this is a single-commit pull request, if you want to see history of commits, this is the [branch I will delete](https://github.com/ec-europa/europa-component-library/pull/294), but keep only for reference purpose.

# QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] `package.json` is up-to-date and `@ec-europa/ecl-base` is part of the dependencies
- [x] I have given the fractal status “ready” to my component
- [x] I have declared `@define mycomponent` in the SCSS file
- [x] I have specified `margin: 0;` on the CSS component
- [ ] I have provided tests
- [x] I follow the naming guidelines
- [ ] the component supports composition
- [ ] the template is fully functional
- [x] I have filled the README.md file (at least a few lines)
- [x] I have checked the dependencies
- [x] there are no hardcoded strings (all content come from the context)

### CODE REVIEW

File: framework/components/ecl-dialogs/ecl-dialogs.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  margin: 0;
+  padding: map-get($ecl-spacing, m);
+  position: fixed;
+  text-align: center;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 90%;
+  z-index: map-get($ecl-z-index, 'modal');
+
+  &[aria-hidden='true'],
```
Comment (yhuard) [inline framework/components/ecl-dialogs/ecl-dialogs.scss]: Can't it be just:

```scss
  display: none;

  &[aria-hidden='false'] {
    display: block;
  }
```

**Review by yhuard — CHANGES_REQUESTED** [review]
A bit hard to test, maybe adding a new preview template could help. 

By the way, the focus is not captured when the modal is open (I can focus elements outside of the modal). Furthermore, I'd give the focus to the first focusable element inside the modal when you open it.

Maybe it's time to discuss INNO-626 ;)

File: framework/components/ecl-dialogs/ecl-dialogs.scss (hunk 1 above)
Comment (kalinchernev) [inline framework/components/ecl-dialogs/ecl-dialogs.scss]: I think it'd be even better to reuse the global reset approach. Thanks for pointing out a possible improvement!

**Comment by kalinchernev** [comment]
Modifications applied.

**Comment by yhuard** [comment]
I'll check

File: framework/components/ecl-dialogs/dialogs.js [hunk 2]
```diff
@@ -43,16 +43,17 @@ export const dialogs = (
   const focusableElements = [].slice.call(
     queryAll(
```
Comment (yhuard) [inline framework/components/ecl-dialogs/dialogs.js]: Better:

```js
const dialogWindow = document.getElementById(dialogWindowId);
const focusableElements =  queryAll('a[href], ..........', dialogWindow);
```

**Review by yhuard — CHANGES_REQUESTED** [review]

File: framework/components/ecl-dialogs/dialogs.js (hunk 2 above)
Comment (kalinchernev) [inline framework/components/ecl-dialogs/dialogs.js]: right, thanks!

File: framework/components/ecl-dialogs/dialogs.js [hunk 3]
```diff
@@ -125,10 +126,14 @@ export const dialogs = (
     firstFocusableElement.focus();
 
     // Close dialog when clicked out of the dialog window.
-    dialogOverlay.addEventListener('click', close);
+    dialogOverlay.addEventListener('click', () => {
```
Comment (yhuard) [inline framework/components/ecl-dialogs/dialogs.js]: May I ask why the change?

Comment (kalinchernev) [inline framework/components/ecl-dialogs/dialogs.js]: Just to make it shorter.

Comment (yhuard) [inline framework/components/ecl-dialogs/dialogs.js]: Sure! You can apply the same logic to line 129, just to be consistent ;)

Comment (kalinchernev) [inline framework/components/ecl-dialogs/dialogs.js]: Right ;D
PS Sorry I missed it

**Review by yhuard — APPROVED** [review]

**Comment by yhuard** [comment]
Works like a charm!

**Pull request merged**